docs/topics/data-analysis/data-analysis-work-with-api.md

```diff
@@ -1,16 +1,176 @@
+df
+```
+
+6. **Concatenate items and display the resulting data frame:** Finally, extract and concatenate items from the DataFrame:
+
+```kotlin
+val items = df.items.concat()
+items
+```
+
+## Clean and refine data
+
+Cleaning and refining data are crucial steps in preparing your dataset for analysis. The [Kotlin DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html) 
+offers powerful functionalities for these tasks. Functions like [`.move`](https://kotlin.github.io/dataframe/move.html), 
+[`.concat`](https://kotlin.github.io/dataframe/concatdf.html), [`.select`](https://kotlin.github.io/dataframe/select.html), 
+[`.parse`](https://kotlin.github.io/dataframe/parse.html), and [`.join`](https://kotlin.github.io/dataframe/join.html) 
+are instrumental in organizing and transforming your data. 
+
+Let's explore an example where we have already [fetched some data using YouTube's data API](#fetch-data-from-an-api).
+The goal is to clean and restructure the dataset to prepare for in-depth analysis:
+
+1. **Reorganize and clean data:**
+You can start by reorganizing and cleaning your data. Here, we're moving certain columns under new headers and removing unnecessary ones for clarity:
+
+```kotlin
+val videos = items.dropNulls { id.videoId }
+    .select { id.videoId named "id" and snippet }
+    .distinct()
+videos
+```
+
+2. **Chunk and load data:**
+Chunk IDs from the cleaned data and load corresponding video statistics. This involves breaking the data into smaller 
+batches and fetching additional details:
+
+```kotlin
+val statPages = clean.id.chunked(50).map {
+    val ids = it.joinToString("%2C")
+    load("videos?part=statistics&id=$ids")
+}
+statPages
+```
+
+3. **Concatenate and select relevant data:**
+Concatenate the fetched statistics and select relevant columns:
+
+```kotlin
+val stats = statPages.items.concat().select { id and statistics.all() }.parse()
+stats
+```
+
+4. **Join DataFrames:**
+Join the existing cleaned data with the newly fetched statistics. This merges two sets of data into a comprehensive DataFrame:
+
+```kotlin
+val joined = clean.join(stats)
+joined
+```
+
+This example demonstrates how to clean, reorganize, and enhance your dataset using Kotlin DataFrame's various functions. 
+Each step is designed to refine the data, making it more suitable for [in-depth analysis](#analyze-data-in-kotlin-notebook).
+
+## Analyze data in Kotlin Notebook
+
+After you've successfully [fetched](#fetch-data-from-an-api) and [cleaned and refined your data](#clean-and-refine-data) 
+using functions from the [Kotlin DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html), the next step 
+is to analyze this prepared dataset to extract meaningful insights.
+
+Functions such as [`.groupBy`](https://kotlin.github.io/dataframe/groupby.html) for categorizing data, 
+[`.sum`](https://kotlin.github.io/dataframe/sum.html) and [`.maxBy`](https://kotlin.github.io/dataframe/maxby.html) for 
```

Comment:
maxBy, sum can be referred as summary statistics. There's good overview in our documentation about them
https://kotlin.github.io/dataframe/summarystatistics.html